tests/test_integration_async.py

```diff
@@ -0,0 +1,617 @@
+        with self.settings(INTERNAL_IPS=[]):
+            response = await self.async_client.get(url, data)
+            self.assertEqual(response.status_code, 404)
+            response = await self.async_client.get(
+                url, data, headers={"x-requested-with": "XMLHttpRequest"}
+            )
+            self.assertEqual(response.status_code, 404)
+
+    async def test_template_source_errors(self):
+        url = "/__debug__/template_source/"
+
+        response = await self.async_client.get(url, {})
+        self.assertContains(
+            response, '"template_origin" key is required', status_code=400
+        )
+
+        template = get_template("basic.html")
+        response = await self.async_client.get(
+            url,
+            {"template_origin": signing.dumps(template.template.origin.name) + "xyz"},
+        )
+        self.assertContains(response, '"template_origin" is invalid', status_code=400)
+
+        response = await self.async_client.get(
+            url, {"template_origin": signing.dumps("does_not_exist.html")}
+        )
+        self.assertContains(response, "Template Does Not Exist: does_not_exist.html")
+
+    async def test_sql_select_checks_show_toolbar(self):
+        url = "/__debug__/sql_select/"
+        data = {
+            "signed": SignedDataForm.sign(
+                {
+                    "sql": "SELECT * FROM auth_user",
+                    "raw_sql": "SELECT * FROM auth_user",
+                    "params": "{}",
+                    "alias": "default",
+                    "duration": "0",
+                }
+            )
+        }
+
+        response = await self.async_client.post(url, data)
+        self.assertEqual(response.status_code, 200)
+        response = await self.async_client.post(
+            url, data, headers={"x-requested-with": "XMLHttpRequest"}
+        )
+        self.assertEqual(response.status_code, 200)
+        with self.settings(INTERNAL_IPS=[]):
+            response = await self.async_client.post(url, data)
+            self.assertEqual(response.status_code, 404)
+            response = await self.async_client.post(
+                url, data, headers={"x-requested-with": "XMLHttpRequest"}
+            )
+            self.assertEqual(response.status_code, 404)
+
+    async def test_sql_explain_checks_show_toolbar(self):
+        url = "/__debug__/sql_explain/"
+        data = {
+            "signed": SignedDataForm.sign(
+                {
+                    "sql": "SELECT * FROM auth_user",
+                    "raw_sql": "SELECT * FROM auth_user",
+                    "params": "{}",
+                    "alias": "default",
+                    "duration": "0",
+                }
+            )
+        }
+
+        response = await self.async_client.post(url, data)
+        self.assertEqual(response.status_code, 200)
+        response = await self.async_client.post(
+            url, data, headers={"x-requested-with": "XMLHttpRequest"}
```

Comment:
By the way, the `x-requested-with` header is not needed anymore. Django has removed `request.is_ajax()` and we do not depend on it either.

Reply:
Should I remove them from both integration tests?

Reply:
Remove them from the new tests for sure, but maybe a different pull request for the existing tests to keep this PR small(er). Thanks!